docs/design/mono/components.md

```diff
@@ -11,6 +11,33 @@ linking).
 To that end this document describes a methodology for defining and implementing components and for calling component
 functionality from the runtime.
 
+## Possible scenarios
```

Comment:
I think another "goal" here would be to reduce the number of build configurations. An argument could be made that all of the below can be accomplished through the `FEATURE_*` build approach in CoreCLR. I think a benefit of the approach here is the componentization may actually reduce the build complexity and instead allow composition closer to run time instead of requiring customized builds. A cons of this approach is that we lose out on the possible benefits of link time code gen.

Reply:
yea one (unstated - but I'll add it) assumption here is that we can push off the final linking (on static linked workloads, anyway) to the workload.  In effect the runtime will give you lego pieces (static libs) and an instruction manual (.props and .targets) and you get to put the final spaceship together.